text/0000-edition-2021-and-beyond.md

```diff
@@ -0,0 +1,194 @@
+- Feature Name: N/A
+- Start Date: 2020-07-29
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+* Announce plans for a Rust 2021 Edition, and for a regular cadence of editions every 3 years thereafter.
+    * We will roll out an edition regardless of whether there are breaking changes.
+* Unlike Rust 2018, we will avoid using editions as a "deadline" to tie together high-priority projects.
+    * Instead, we embrace the train model, but editions are effectively a "somewhat bigger release", giving us an opportunity to give an overview of all the work that has landed over the previous three years.
+* We specify a cadence for Edition lints.
+    * "Edition idiom" lints for Edition N will warn for editions before N, and become "deny by default" in Edition N.
+    * Since it would be disruptive to introduce deny-by-default lints for Rust 2018 now, the Rust 2018 lints are repurposed into Rust 2021 Edition lints.
+* We specify a policy on reserving keywords and other prospective changes.
+    * In short, reserving keywords is allowed only as part of an active project group.
+
+# Motivation
+[motivation]: #motivation
+
+The plan for editions was laid out in [RFC 2052] and Rust had its first edition in 2018. This effort was in many ways a success but also resulted in some difficult lessons. As part of this year's roadmap, one of the major questions we identified was that we need to decide whether we are going to do more editions and -- if so -- how we are going to manage the process.
+
+[RFC 2052]: https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md
+
+This RFC proposes a specific answer to those questions:
+
+* We will do new Rust editions on a regular, three-year cadence.
+    * We will roll out an edition regardless of whether there are breaking changes.
+* Unlike Rust 2018, we will avoid using editions as a "deadline" to tie together high-priority projects.
+    * Instead, we embrace the train model, but editions are effectively a "somewhat bigger release", giving us an opportunity to give an overview of all the work that has landed over the previous three years.
+* We specify a cadence for Edition lints.
+    * "Edition idiom" lints for Edition N will warn for editions before N, and become "deny by default" in Edition N.
+    * Since it would be disruptive to introduce deny-by-default lints for Rust 2018 now, the Rust 2018 lints are repurposed into Rust 2021 Edition lints.
+* We specify a policy on reserving keywords and other prospective changes.
+    * In short, reserving keywords is allowed only as part of an active project group.
```

Comment:
Sorry to be picky, but this are no questions.
Probably you also want to link to the [questions](https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md#unresolved-questions) and copy them here?
Are this the open questions you are referring to or did I miss anything?

1. What impact is there, if any, on breakage permitted today for bug fixing or soundness holes? In many cases these are more disruptive than introducing a new keyword.
2. Is "edition" the right key in Cargo.toml? Would it be more clear to just say rust = "2019"?
3. Will we ever consider dropping support for very old editions? Given the constraints in this RFC, it seems unlikely to ever be worth it.
4. Should rustc default to the latest edition instead?
5. How do we handle macros, particularly procedural macros, that may mix source from multiple editions?

Question 4 should be reformulated to fit in this context, since an implicit answer is given.

Reply:
Updating the RFC to include this stuff is probably a good idea, but some of this was set in the original editions RFC, and the intention isn't really to change that.

1. this isn't edition specific; we already reserve the right to fix soundness holes without an edition.
2. given that we already have the edition key, and adding a new one would add even *more* back-compat stuff to deal with, i personally believe sticking with edition is correct. "rust" was considered and rejected during the 2018 edition process.
3. there is no current plan to, and the design makes it such that supporting them shouldn't be a burden.
4. this would be a breaking change, and so is not likely. you could argue that a rustc2.0 would be fine, but given that we don't have a full language spec and multiple production ready compilers, this seems unwise.
5. this already Just Works, as far as I know.

Reply:
Oh, and that previous RFC: https://github.com/rust-lang/rfcs/blob/master/text/2052-epochs.md

Reply:
Those questions from the original RFC were actually not the unresolved questions we are trying to answer, more like "general questions that arose in and after the edition". Some of them may have been listed in the RFC, I'm not sure. I'll rephrase perhaps.